### PR TITLE
Delete cloud database support

### PR DESCRIPTION
**Pull request by nasark**
- added support to delete databases on oracle cloud
- refactored spec tests to better differentiate oracle and mysql database actions

@miq-bot assign @agrare 
@miq-bot add_label enhancement

### CODE REVIEW

File: app/models/manageiq/providers/oracle_cloud/cloud_manager/cloud_database.rb [hunk 1]
```diff
@@ -164,4 +165,23 @@ def self.create_mysql_database(ext_management_system, options)
       )
     end
   end
+
+  def raw_delete_cloud_database
+    db_engine.include?("Oracle Database") ? delete_oracle_database : delete_mysql_database
```
Comment (nasark) [inline app/models/manageiq/providers/oracle_cloud/cloud_manager/cloud_database.rb:170]: Seemed like the best way to differentiate between the two types of databases based on https://github.com/ManageIQ/manageiq-providers-oracle_cloud/blob/master/app/models/manageiq/providers/oracle_cloud/inventory/parser.rb#L55-L74

Comment (agrare) [reply app/models/manageiq/providers/oracle_cloud/cloud_manager/cloud_database.rb:170]: Agree this is the best way to differentiate, could we pull this out into its own method like `db_type` and return `:mysql` or `:oracle` ?

Comment (nasark) [reply app/models/manageiq/providers/oracle_cloud/cloud_manager/cloud_database.rb]: Sure, I'll make the change

**Review by kbrock — APPROVED** [review]
this looks nice.

Pulling that into `db_type` was a nice suggestion/change.

**Pull request merged**